modules/util/time_str.go

```diff
@@ -0,0 +1,99 @@
+// Copyright 2022 Gitea. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+package util
+
+import (
+	"fmt"
+	"math"
+	"regexp"
+	"strconv"
+	"strings"
+)
+
+var (
+	// Time estimate match regex
+	rTimeEstimateOnlyHours = regexp.MustCompile(`^([\d]+)$`)
+	rTimeEstimateWeeks     = regexp.MustCompile(`([\d]+)w`)
+	rTimeEstimateDays      = regexp.MustCompile(`([\d]+)d`)
+	rTimeEstimateHours     = regexp.MustCompile(`([\d]+)h`)
+	rTimeEstimateMinutes   = regexp.MustCompile(`([\d]+)m`)
+)
+
+// TimeEstimateFromStr returns time estimate in seconds from formatted string
+func TimeEstimateFromStr(timeStr string) int64 {
```

Comment:
adding tests for this function would be awesome!

Reply:
Will work on it